harvester/records/record.py

```diff
@@ -616,3 +616,46 @@ def get_single_subfield(tag: DataField, subfield: str) -> SubField | None:
             message = f"Multiple subfields found in tag for subfield: {subfield}"
             raise ValueError(message)
         return None
+
+    def get_single_tag_subfield_value(self, tag_code: str, subfield_code: str) -> str:
+        """Return single subfield value from a single tag."""
+        tag = self.get_single_tag(tag_code)
+        if not tag:
+            message = f"Record does not have single instance of tag '{tag_code}'"
+            raise ValueError(message)
+        subfield = self.get_single_subfield(tag, subfield_code)
+        if not subfield:
+            message = f"Tag does not have single instance of subfield '{subfield_code}'"
+            raise ValueError(message)
+        return subfield.value.strip()
+
+    def get_multiple_tag_subfield_values(
```

Comment:
Good question.  I think it actually relies more on the behavior of the `marcalyx` library, where `Record.field("XXX")` will _always_ return a list.

Often, we want the value from a non-repeating (NR) field and one subfield.  In this case, `get_single_tag_and_subfield_value()` would make sense.  However, _technically_ it would work on a repeating (R) field that may only have one instance in the record.

These convenience methods still need to be used with some understanding and caution, but the benefit is that `get_single...()` will throw an exception if more than one instance exists.